### DISCUSSION TITLE
Is it possible to run Patchman-Client on a container?

### DISCUSSION THREAD

**Discussion by Ezio1001** [furlongm/patchman#560] [Q&A]
Sorry If my questions sound dumb to you guys, I'm still new.
I've been wondering about the possibility of running the client over a container.
Few questions came to mind, first would be how the client would collect reports? Would it collect reports from the container it's deployed within? since that sounds like the logical outcome. 
The second question would be what is the process that takes place exactly when running `patchman-client`? Does it collect from a specific directory or multiple different ones? since that seems like the case with it.
And finally, is there a way for patchman-client to bypass the container and collect data from the host that the container is hosted within?

**Comment by TW1920** [comment]
Should be working - not tested, but why not. You only need the create a container at your own and install it there manually or creating own docker container for patchman.

The reports will be sent by the patchman-client (Bash script) to the patchman server. The final processing is running on the patchman server.

The patchman client need to be installed on every host where you want to get reports. It's automatically running after apt update, you can set up manually it's running after booting and you could create a cronjob.

Yeah, still same way like every other host... To get patchman reachable you can set the http/https ports directly to the patchman container or you use a nginx reverse proxy to redirect to the container. Other way would be other ports than 80/4443, than you could do it without reverse proxy - don't forget to configure the /etc/patchman/patchman-client.conf (Client - setting the server, ports if not 80/443...)

For the server you will find the vonf here: /etc/patchman/local_settings.py

**Reply by Ezio1001** [reply]
Actually this question was before I made it work.
The way I went about this approach is I created my own docker file, minimized the unnecessary installations like SQL server and patchman client (I'm talking about patchman only as the server), then built my container from that docker file, had the DB on a separate container and after making the necessary changes linked the server to the DB and got it up and running.
As for the patchman-client, I extracted it since it's a bash script and I'm running it as a bash script over the required hosts in order to not add a repo and install from it, this makes automation approach easier.
